compiler-core/src/analyse.rs

```diff
@@ -644,17 +647,9 @@ fn infer_function(
             expr_typer.infer_fn_with_known_types(args_types, body, Some(return_type))?;
         let args_types = args.iter().map(|a| a.type_.clone()).collect();
         let typ = fn_(args_types, body.last().type_());
-        Ok((typ, args, body, expr_typer.supported_targets))
+        Ok((typ, args, body, expr_typer.implementations))
     })?;
 
-    if is_placeholder {
-        // If the function has an empty body we are only going to consider as
-        // supported targets the ones that have an external implementation.
-        supported_targets = external_targets;
-    } else {
-        supported_targets = supported_targets.merge(external_targets);
-    }
-
```

Comment:
Just a bit of explanation on these changes to make it easier to follow: this piece of target tracking logic was moved from here inside the expression typer. This way everything is done by the typer and the target tracking logic is not split in two different places

Reply:
Thank you, super helpful!